tests/rbd_mirror/test_rbd_mirror_image_features.py

```diff
@@ -0,0 +1,211 @@
+        # Disable image features in primary cluster
+        mirror1.image_feature_disable(imagespec=imagespec, image_feature=image_feature)
+        mirror2.resync(imagespec=imagespec)
+        # wait till image gets reflected to secondary
+        time.sleep(60)
+        # Verify image feature reflection in secondary cluster
+        out = rbd2.image_info(poolname, imagename)
+        for word in search_pattern:
+            if word not in out["flags"]:
+                log.error("Image features not disabled in secondary")
+                return 1
+        log.info("Image features got disabled successfully in secondary")
+
+        # Enable image features in primary cluster
+        mirror1.image_feature_enable(imagespec=imagespec, image_feature=image_feature)
+        mirror2.resync(imagespec=imagespec)
+        time.sleep(60)
+        # Verify image feature reflection in secondary cluster
+        out = rbd2.image_info(poolname, imagename)
+        for word in search_pattern:
+            if word in out["flags"]:
+                log.error("Image features not enabled in secondary site")
+                return 1
+        log.info("Image features got enabled successfully in secondary")
+        return 0
+
+    except ValueError as ve:
+        log.error(
+            f"{kw.get('ceph_cluster_dict').values} has less or more clusters Than Expected(2 clusters expected)"
+        )
+        log.exception(ve)
+        return 1
+    except Exception as e:
+        log.exception(e)
+        return 1
+    finally:
+        mirror1.clean_up(peercluster=mirror2, pools=[poolname])
+
+
+def run(**kw):
+    """Verification of change in properties of image in Primary Pool
+    gets reflected on remote site upon modification.
+
+    Args:
+        **kw:
+    Returns:
+        0 - if test case pass
+        1 - it test case fails
+
+    Test case covered -
+    CEPH-9520 - When a both Pool and image based mirroring is established,
+    verify if change of any image features in primary site should reflect
+    on remote site upon modification.
+
+    Test Case Flow:
+    1. Create a pool on both clusters.
+    2. Configure mirroring (peer bootstrap) between two clusters.
+    3. Enable pool mode journal based mirroring on the pool respectively.
+    4. Verify if change of any image features in primary site gets mirrored in secondary site.
+    5. Enable image mode snapshot based mirroring on the pool respectively.
+    6. Verify if change of any image features in primary site gets mirrored in secondary site.
+    """
+    log.info(
+        "Starting CEPH-9520 [Rbd Mirror] - Verify if change in properties of \
+        image in Primary Pool gets reflected on remote site upon modification "
+    )
+    mirror_obj = rbd_mirror_config(**kw)
+    # To test against pool based journal mirror mode
+    if mirror_obj:
+        log.info("Executing test on replicated pool")
+        if test_journal_image_feature(
```

Comment:
use a iterator to run `test_journal_image_feature` and `test_snapshot_image_feature`.

Reply:
done

Reply:
I do not see any iterator being used to run these, @sunilangadi2 when you say done what are the changes you did for this comment?
cc: @sunilkumarn417